### DISCUSSION TITLE
Advisories table: which CVE is associated to which product?

### DISCUSSION THREAD

**Discussion by carlosthe19916** [trustification/trustification#227] [Announcements]
When we enter to the Advisories table and expand the content of any row we can see:
- On the left a list of CVEs associated with the advisory.
- On the Right, a list of products affected by the CVEs listed on the left side.

How can we know which CVE is associated to any product (the right side of the expanded table)?
- Should I interpret the table as: "all products have all CVEs listed" OR "some products have some CVEs listed"?
- As a user it would be beneficial to understand this relationship.

Common questions when it comes to CVEs might come from 2 sources:
- My team is responsible for only 1 product therefore I'd like to understand exactly which CVEs affects to my product, I'm not interested in other products.
- There is vulnerability discovered like "Lo4J" and I'd like to know all products affected by this CVE.

The reasoning in both cases above start from "I have 1 product => I want to list 'N' CVEs" or "I have 1 CVE in mind and then I want to list all products affected". The current way of presenting the data looks like "I have N products and I can see N CVEs"

![Screenshot from 2023-07-04 09-45-12](https://github.com/trustification/trustification/assets/2582866/51b1c2f3-c3af-4750-af14-08b93a7850b1)

I didn't want to create an issue since I'm not even sure my thoughts are reasonable or if the data supports this.
My intention is just to raise a question and if appropriate find a better way of presenting this data to the user.

**Comment by ctron** [comment]
Step 1, it might be helpful to get a more complex view on this (including source view). I am working in this.

In any case, you are right that this isn't very user friendly. There is most likely some more information in there, but not much. A CSAF document like this covers a bunch of things. And depending on the information we have, that might be enough or confusing.

**Comment by lulf** [comment]
The CSAF document should contain, for each vulnerability, the product id where vuln is fixed, affected etc. The id can be used to look into the 'relationships' table to find a key to look into the 'product_tree' to find the exact identifiers to which the vulnerability status applies.

This process is implemented here https://github.com/trustification/trustification/blob/main/vexination/index/src/lib.rs#L445 - but could perhaps be reused in the frontend for making some meaningful grouping of the information.
